src/query/functions/tests/it/scalars/testdata/datetime.txt

```diff
@@ -3345,3 +3345,291 @@ output domain  : {1630812366000000..=1630812366000000}
+raw expr       : diff_years(to_timestamp('2025-03-27 21:01:35.423179'), to_timestamp('2023-11-12 09:38:18.165575'))
+checked expr   : diff_years<Timestamp, Timestamp>(to_timestamp<String>("2025-03-27 21:01:35.423179"), to_timestamp<String>("2023-11-12 09:38:18.165575"))
+optimized expr : 1_i64
+output type    : Int64
+output domain  : {1..=1}
+output         : 1
+
+
+ast            : date_diff(year, to_timestamp('2020-02-29 23:59:59.165575'), to_timestamp('2019-02-28 23:59:59.423179'))
+raw expr       : diff_years(to_timestamp('2019-02-28 23:59:59.423179'), to_timestamp('2020-02-29 23:59:59.165575'))
+checked expr   : diff_years<Timestamp, Timestamp>(to_timestamp<String>("2019-02-28 23:59:59.423179"), to_timestamp<String>("2020-02-29 23:59:59.165575"))
+optimized expr : 0_i64
+output type    : Int64
+output domain  : {0..=0}
+output         : 0
+
+
+ast            : date_diff(quarter, to_date(0), to_date(10000))
+raw expr       : diff_quarters(to_date(10000), to_date(0))
+checked expr   : diff_quarters<Date, Date>(to_date<Int64>(to_int64<UInt16>(10000_u16)), to_date<Int64>(to_int64<UInt8>(0_u8)))
+optimized expr : 109_i64
+output type    : Int64
+output domain  : {109..=109}
+output         : 109
+
+
+ast            : date_diff(quarter, to_date(10000), to_date(0))
+raw expr       : diff_quarters(to_date(0), to_date(10000))
+checked expr   : diff_quarters<Date, Date>(to_date<Int64>(to_int64<UInt8>(0_u8)), to_date<Int64>(to_int64<UInt16>(10000_u16)))
+optimized expr : -109_i64
+output type    : Int64
+output domain  : {-109..=-109}
+output         : -109
+
+
+ast            : date_diff(quarter, to_date('2000-01-01'), to_date('2024-12-31'))
+raw expr       : diff_quarters(to_date('2024-12-31'), to_date('2000-01-01'))
+checked expr   : diff_quarters<Date, Date>(to_date<String>("2024-12-31"), to_date<String>("2000-01-01"))
+optimized expr : 99_i64
+output type    : Int64
+output domain  : {99..=99}
+output         : 99
+
+
+ast            : date_diff(quarter, to_timestamp('2023-11-12 09:38:18.165575'), to_timestamp('2025-03-27 21:01:35.423179'))
+raw expr       : diff_quarters(to_timestamp('2025-03-27 21:01:35.423179'), to_timestamp('2023-11-12 09:38:18.165575'))
+checked expr   : diff_quarters<Timestamp, Timestamp>(to_timestamp<String>("2025-03-27 21:01:35.423179"), to_timestamp<String>("2023-11-12 09:38:18.165575"))
+optimized expr : 5_i64
+output type    : Int64
+output domain  : {5..=5}
+output         : 5
+
+
+ast            : date_diff(quarter, to_timestamp('2020-02-29 23:59:59.165575'), to_timestamp('2019-02-28 23:59:59.423179'))
+raw expr       : diff_quarters(to_timestamp('2019-02-28 23:59:59.423179'), to_timestamp('2020-02-29 23:59:59.165575'))
+checked expr   : diff_quarters<Timestamp, Timestamp>(to_timestamp<String>("2019-02-28 23:59:59.423179"), to_timestamp<String>("2020-02-29 23:59:59.165575"))
+optimized expr : -3_i64
+output type    : Int64
+output domain  : {-3..=-3}
+output         : -3
```

Comment:
I test in snowflake , `select datediff(quarter, to_timestamp('2020-02-29 23:59:59.165575'), to_timestamp('2019-02-28 23:59:59.423179'))` expect -4